packages/main/src/components/DynamicPage/index.tsx

```diff
@@ -0,0 +1,229 @@
+          },
+          { once: true }
+        );
+      });
+    },
+    [dynamicPageRef, classes.headerCollapsed, headerContentHeight, topHeaderHeight, headerPinned]
+  );
+
+  const onHoverToggleButton = useCallback(
+    (e) => {
+      if (e && e.type === 'mouseover') {
+        // TODO background color should be sapObjectHeader_Hover_Background (same color as sapTile_Active_Background)
+        topHeaderRef.current.style.backgroundColor = ThemingParameters.sapTile_Active_Background;
+      } else {
+        topHeaderRef.current.style.backgroundColor = null;
+      }
+    },
+    [topHeaderRef]
+  );
+
+  const onToggleHeaderContent = (e) => {
+    if (e.target.tagName === 'DIV') {
+      onToggleHeaderContentVisibility(e, anchorBarRef.current.children.item(0));
+    }
+  };
+
+  let currentIcon = 'slim-arrow-up';
+  dynamicPageRef.current?.addEventListener('scroll', () => {
+    if (
+      anchorBarRef.current?.children?.[0].icon === 'slim-arrow-up' &&
+      anchorBarRef.current?.children?.[0].icon !== currentIcon
+    ) {
+      setHeaderVisible(true);
+      currentIcon = 'slim-arrow-up';
+      anchorBarRef.current.style.top = '0.025rem';
+    } else if (
+      anchorBarRef.current?.children?.[0].icon === 'slim-arrow-down' &&
+      anchorBarRef.current?.children?.[0].icon !== currentIcon
+    ) {
+      setHeaderVisible(false);
+      currentIcon = 'slim-arrow-down';
+      anchorBarRef.current.style.top = '-0.025rem';
+    }
+  });
+
+  useEffect(() => {
+    if (headerPinned === true) {
+      anchorBarRef.current.style.top = '0.025rem';
+    }
+  }, [headerPinned]);
+
+  return (
+    <div
+      ref={dynamicPageRef}
+      title={tooltip}
+      className={dynamicPageClasses.toString()}
+      style={style}
+      {...passThroughProps}
+    >
+      {cloneElement(title, {
+        ref: topHeaderRef,
+        headerVisible,
+        onToggleHeaderContentVisibility: onToggleHeaderContent
+      })}
+      {cloneElement(header, {
+        ref: headerContentRef,
+        style: { top: noHeader ? 0 : topHeaderHeight },
+        headerPinned,
+        topHeaderHeight,
+        classes
+      })}
+      <FlexBox
+        className={classes.anchorBar}
+        style={{
+          top: headerPinned
+            ? topHeaderRef.current?.scrollHeight + headerContentRef.current?.scrollHeight
+            : topHeaderRef.current?.scrollHeight
+        }}
+      >
+        <DynamicPageAnchorBar
+          ref={anchorBarRef}
+          style={{ top: '0.025rem' }}
+          headerContentPinnable={headerContentPinnable}
+          showHideHeaderButton={showHideHeaderButton && !noHeader}
+          headerContentHeight={headerContentHeight}
+          onToggleHeaderContentVisibility={onToggleHeaderContentVisibility}
+          setHeaderPinned={setHeaderPinned}
+          headerPinned={headerPinned}
+          onHoverToggleButton={onHoverToggleButton}
+        />
+      </FlexBox>
+      {children ? <div ref={childrenRef}>{children}</div> : null}
```

Comment:
The childrenRef is not used at all and can be removed
```suggestion
      {children}
```